src/storage/db_storage.cpp

```diff
@@ -281,52 +287,155 @@ std::map<blk_hash_t, bool> DbStorage::getAllDagBlockState() {
   return res;
 }
 
+void DbStorage::savePeriodData(const PbftBlock& pbft_block, const std::vector<Vote>& cert_votes,
+                               const std::vector<DagBlock>& dag_blocks, const std::vector<Transaction>& transactions,
+                               Batch& write_batch) {
+  uint64_t period = pbft_block.getPeriod();
+  addPbftBlockPeriodToBatch(period, pbft_block.getBlockHash(), write_batch);
+
+  // Add dag_block_period in DB
+  uint64_t block_pos = 0;
+  for (auto const& blk : dag_blocks) {
+    addDagBlockPeriodToBatch(blk.getHash(), period, block_pos, write_batch);
+    block_pos++;
+  }
+
+  RLPStream s;
+  s.appendList(4);
+  s.appendRaw(pbft_block.rlp(true));
+  s.appendList(cert_votes.size());
+  for (auto const& vote : cert_votes) {
+    s.appendRaw(vote.rlp());
+  }
+  s.appendList(dag_blocks.size());
+  std::vector<Slice> dag_blks_to_remove;
+  std::vector<Slice> trxs_to_remove;
+  dag_blks_to_remove.reserve(dag_blocks.size());
+  for (auto const& block : dag_blocks) {
+    dag_blks_to_remove.emplace_back(toSlice(block.getHash()));
+    s.appendRaw(block.rlp(true));
+  }
+  s.appendList(transactions.size());
+  uint64_t position = 0;
+  trxs_to_remove.reserve(transactions.size());
+  for (auto const& trx : transactions) {
+    trxs_to_remove.emplace_back(toSlice(trx.getHash()));
+    addTransactionStatusToBatch(write_batch, trx.getHash(),
+                                TransactionStatus(TransactionStatusEnum::executed, period, position));
+    s.appendRaw(*trx.rlp());
```

Comment:
so now we do not save pre-cached transaction "verification" as we used to in function `void saveTransaction(Transaction const& trx, bool verified = false)` that is used in finalize function. we have to save it as before. So I would remove it from save function and have it here but only when we actually verify them before (not sure if it is happening for while syncing)